### DISCUSSION TITLE
Multiphase tessellation

### DISCUSSION THREAD

**Discussion by LeiYulut** [neperfepx/neper#726] [Q&A]
Hi,
I'm tessellating a multiphase material of 4 phases with similar grain size and uniform distribution. Could I use the option "-group" to distinguish them? The microstructure morphology is shown below:
![microstructure](https://github.com/neperfepx/neper/assets/140691482/10648f7f-008c-4589-9942-6b7a60343bd0)

I have tried to use volume to control their distribution, but could not realize my goal well. Is there any other method to realize this?

Thanks a lot

**Comment by rquey** [comment]
Since, by default, the cells are randomly distributed (spatially and also size-wise), you can simply group them by ids. Assuming all phases represent the same volume fraction:
```
neper -T -n 1000 -dim 2 -group "(id<=250)?1:((id<=500)?2:((id<=750)?3:4))"
neper -V n1000-id1.tess -datacellcol group -print img
```
![image](https://github.com/neperfepx/neper/assets/2659376/71ab3b1a-8e19-47ed-bbc4-74607e25a5f0)

(You can use the `-morpho` option as you'd like.)

Otherwise, sum several `diameq` (identical, in your case) distributions and create groups based on the "modes":
```
neper -T -n 1000 -dim 2 -morpho "diameq:lognormal(1,0.42)+lognormal(1,0.42)+lognormal(1,0.42)+lognormal(1,0.42),1-circularity:lognormal(0.1,0.03)" -group mode
neper -V n1000-id1.tess -datacellcol group -print img
```

![image](https://github.com/neperfepx/neper/assets/2659376/6b9d698b-36fa-4684-a227-f65368f3188f)

**Reply by LeiYulut** [reply]
Dear Rquey,
Thank you for your reply. I will try it.
Thanks a lot
